### PR TITLE
Board summary api

### PR DESCRIPTION
**Pull request by cftang0827**
- Add `boardsummary` in ptt board_list.go & bbs and api layer
- Register board summary endpoint in main.go
- TODO: Add unit test

### CODE REVIEW

**Comment by cftang0827** [comment]
I will try to put some unit tests in this PR

**Comment by chhsiao1981** [comment]
hi @cftang0827 ～

Thanks for the PR～
It looks like that you forgot including bbs.LoadBoardSummary？～

File: ptt/board_list.go [hunk 1]
```diff
@@ -8,6 +8,50 @@ import (
...
+// 	uid
+//	bid
+
+//Return:
+//	summary
+//	err
+func LoadBoardSummary(user *ptttype.UserecRaw, uid ptttype.Uid, bid ptttype.Bid) (summary *ptttype.BoardSummaryRaw, err error) {
+	nBoardsInCache := cache.NumBoards()
+	var boardStat *ptttype.BoardStat
+	for currentIdx := ptttype.SortIdx(0); ; currentIdx++ {
```
[the remark this answers is not in the document]
Comment (cftang0827) [reply ptt/board_list.go:26]: Thanks @chhsiao1981 , I will have a look and try to use O(1) solution to get Board summary info!

File: bbs/load_board_summary.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"fmt"
+
+	"github.com/Ptt-official-app/go-pttbbs/ptt"
+)
+
+func LoadBoardSummary(uuserID UUserID, bboardID BBoardID) (summary *BoardSummary, err error) {
+	bid, bbb, err := bboardID.ToRaw()
+	fmt.Println(string(bbb[:]))
```
Comment (chhsiao1981) [inline bbs/load_board_summary.go:11]: hi @cftang0827 ～

You can use

bid, _, err := bboardID.ToRaw()
to avoid dummy variables.

Comment (cftang0827) [reply bbs/load_board_summary.go:11]: Oh, it's my debug messages, I forgot to remove them. Will clear all debug messages in the code

**Pull request merged**